pkg/validation/usage_groups.go

```diff
@@ -61,8 +67,18 @@ func (c *UsageGroupConfig) GetUsageGroups(tenantID string, lbls phlaremodel.Labe
 	}
 
 	for name, matchers := range c.config {
-		if matchesAll(matchers, lbls) {
-			match.names = append(match.names, name)
+		matches, captureGroups := c.matchesAllWithCaptureGroups(matchers, lbls)
+		if matches {
+			if strings.Contains(name, "$") && len(captureGroups) > 0 {
+				dynamicName := name
+				for i, captureGroup := range captureGroups {
+					placeholder := fmt.Sprintf("$%d", i+1)
+					dynamicName = strings.ReplaceAll(dynamicName, placeholder, captureGroup)
```

Comment:
I guess that would stop working with `$1` and `$10`.

Also there is a bit of second level of replacement depending on the order: Imagine usage group: `$1-$2`

Theoretically the first capture group $1 could also contain a `haha-$2`, that then would be replaced later, but not if it would be `haha-$1`.

You probably need to tokenize it first. Then do the expanding in one go.

More nitpicking: Also probably if someone needs an explicit `$[0-9]` in the label, which has capture groups they might need an escape `$$12`.

Also as source for inspiration might be: https://cs.opensource.google/go/go/+/refs/tags/go1.24.3:src/regexp/regexp.go;l=926